core/src/main/java/org/opensearch/sql/analysis/AnalysisContext.java

```diff
@@ -23,13 +27,27 @@ public class AnalysisContext {
   @Getter
   private final List<NamedExpression> namedParseExpressions;
 
+  /**
+   * Storage for values of functions which return a constant value.
+   * We are storing the values there to use it in sequential calls to those functions.
+   * For example, `now` function should the same value during processing a query.
+   */
+  @Getter
+  @Setter
+  private Map<String, Expression> cachedFunctionValues;
+
   public AnalysisContext() {
     this(new TypeEnvironment(null));
   }
 
+  /**
+   * Class CTOR.
+   * @param environment Env to set to a new instance.
+   */
```

Comment:
`namedParseExpression` and `cachedFunctionValues` can be initialized at declaration and this constructor generated with `@AllArgsConstructor` annotation.